### PR TITLE
Track Fallback on 'Missing' Track Metadata

### PR DESCRIPTION
**Pull request by srbrills**
- Added fallback to master track list (library) if track metadata is missing in playlist.
- Added exception catch for playlists or tracks with special characters.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -84,6 +82,9 @@ def main():
     logging.debug('loading Spotify')
     spot = Spotify_client()
 
+    # Get a full dump of all songs in library
+    library = gpm.gpm_client.get_all_songs()
```
Comment (srbrills) [inline main.py:86]: Get master list of songs in user's library for use later, if needed.

File: main.py [hunk 2]
```diff
@@ -99,10 +100,14 @@ def main():
...
         logging.info(f'Adding tracks to {name}')
-        
+
         for track in playlist.get('tracks'):
+            artist = None
+            title = None
             try:
                 track_data = track.get('track')
+                if not track_data:
+                	track_data = [item for item in library if item['id'] == track['trackId']][0]
```
Comment (srbrills) [inline main.py:110]: If track metadata doesn't exist from playlist data, go find the track in the library and get its information instead.

File: main.py [hunk 3]
```diff
@@ -121,21 +126,33 @@ def main():
...
+            # thrown when 'tracks' is missing
             except AttributeError:
                 logging.info("This track does not have metadata - probably uploaded. Skipping")
-                with open('./errored-tracks.log', 'a') as file:
-                    file.writelines(f'playist: {name}\n')
+                try:
+                    with open('./errored-tracks.log', 'a') as file:
+                        file.writelines(f'playlist: {name}\n\n')
+                except:
+                    logging.info("Error writing playlist to errored list.  It most likely has special characters.")
```
Comment (srbrills) [inline main.py:136]: Noticed that logging would crash the entire script if either the playlist or track had special characters (e.g. korean, japanese, russian, etc.).

**Comment by iloveicedgreentea** [comment]
Thanks for this PR, lgtm!

**Pull request merged**